### PR TITLE
Lambda trigger docs

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
DRAFT PR for lambda trigger docs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cli/triggers.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+If you wish to modify the functionality of these templates, you are able to do so locally before pushing them.
+
+## Auth
+
+The CLI Auth workflow provides the following Lambda trigger templates:
+
+### Custom Auth Challenge with Google reCaptcha
+
+Captchas allow front end applications to guards against bots or other unwanted page interactions by presenting a challenge that is designed to require human intervention.  The Google reCaptcha service is a popular implementation of captcha.
```
Comment (nikhil-dabhade) [inline cli/triggers.md]: to "guard"

Comment (nikhil-dabhade) [inline cli/triggers.md]: Extra space after first sentence

File: cli/triggers.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+The CLI Auth workflow provides the following Lambda trigger templates:
+
+### Custom Auth Challenge with Google reCaptcha
+
+Captchas allow front end applications to guards against bots or other unwanted page interactions by presenting a challenge that is designed to require human intervention.  The Google reCaptcha service is a popular implementation of captcha.  
+
+This template will configure three triggers: CreateAuthChallenge, DefineAuthChallenge and VerifyAuthChallengeResponse.
+
+The first two will essentially allow the standard username/password flow to execute unimpeded, while VerifyAuthChallengeResponse will accept data from your client-side captcha component and verify it against Google's reCaptcha service via axios.
```
Comment (nikhil-dabhade) [inline cli/triggers.md]: We should a link to axios.

File: cli/triggers.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+### Basic Scaffolding for a Custom Auth Challenge
+
+This template will configure three triggers: CreateAuthChallenge, DefineAuthChallenge and VerifyAuthChallengeResponse.
+
+It will not, however, provide a fully-formed custom authentication flow.  Instead, it will create a 'hello world' custom auth flow skeleton that you can manually edit.  The intent of this template is to give you a starting place.
+
+### Email Verification Link with Redirect
+
+Cognito allows you to configure your User Pool to send an email to your users when they attempt to register an account.  You can configure this email to contain a link to Cognito's Hosted UI where the user's account will be marked as confirmed.  However, there is currently no means to redirect the user back to your app from this Hosted UI page.
```
Comment (nikhil-dabhade) [inline cli/triggers.md]: Extra space after first sentence

File: js/authentication.md [hunk 4]
```diff
@@ -86,6 +86,55 @@ A configuration file called `aws-exports.js` will be copied to your configured s
 
 **NOTE**: If your Authentication resources were created with Amplify CLI version 1.6.4 and below, you will need to manually update your project to avoid Node.js runtime issues with AWS Lambda. [Read more]({%if jekyll.environment == 'production'%}{{site.amplify.docs_baseurl}}{%endif%}/cli/lambda-node-version-update)
 
+#### Configuring Cognito Lambda Triggers
+
+The CLI allows you to configure [Lambda Triggers](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-identity-pools-working-with-aws-lambda-triggers.html) for your AWS Cognito User Pool.  These can enable you to add custom functionality to your registration and authentication flows.
```
Comment (nikhil-dabhade) [inline js/authentication.md]: "These enable you to add custom functionality ...."

File: js/authentication.md [hunk 5]
```diff
@@ -86,6 +86,55 @@ A configuration file called `aws-exports.js` will be copied to your configured s
...
+
+There are two ways to setup Lambda Triggers for your Cognito User Pool.
+
+1. In the default Auth CLI workflow, you will be presented with a list of prepared Lambda Trigger templates if you opt to configure advanced settings:
```
Comment (nikhil-dabhade) [inline js/authentication.md]: you will be presented with a list of Lambda Trigger templates

File: cli/triggers.md (hunk 1 above)
Comment (haverchuck) [reply cli/triggers.md]: Thanks :)

File: cli/triggers.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
+# Lamba Triggers
+
+Certain AWS Services can [invoke Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/lambda-services.html) in response to lifecycle events.  The Amplify CLI provides trigger templates for common use cases.
+
+If you wish to modify the functionality of these templates, you are able to do so locally before pushing them.
+
+## Auth
+
```
Comment (undefobj) [inline cli/triggers.md]: I think we need to show a code snippet of how customers surface these templates - e.g. `amplify add auth` then select `configure the custom stuff` etc...

File: cli/triggers.md (hunk 2 above)
Comment (undefobj) [reply cli/triggers.md]: We shouldn't actually mention Axios, instead we should describe what is happening - e.g. when `Auth.sendCustomChallengeAnswer ` runs it invokes the VerifyAuthChallengeResponse in Cognito which calls a Lambda function that executes an HTTP request to Google, thus validating the captcha on the backend and returning the result to the client.

File: cli/triggers.md (hunk 1 above)
Comment (haverchuck) [reply cli/triggers.md]: Fixed.  It's a tough habit to break.

File: cli/triggers.md (hunk 2 above)
Comment (haverchuck) [reply cli/triggers.md]: Done.

File: cli/triggers.md (hunk 3 above)
Comment (haverchuck) [reply cli/triggers.md]: Fixed.

File: js/authentication.md (hunk 5 above)
Comment (haverchuck) [reply js/authentication.md]: I took out 'prepared', but left 'if you opt to configure advanced settings:'.

**Comment by kaustavghosh06** [comment]
@haverchuck 3 things

1. Rebase from the latest master branch and have the triggers file in the `cli-toolchain/` directory instead of the current `cli/` directory
2. Add the custom triggers section and pointer to the trigger sections to the android and ios sections
3. Fix the spellings causing the tests to fail :)

**Pull request closed without merging**